api/src/services/payload.ts

```diff
@@ -293,14 +296,30 @@ export class PayloadService {
 	 * Knex returns `datetime` and `date` columns as Date.. This is wrong for date / datetime, as those
 	 * shouldn't return with time / timezone info respectively
 	 */
-	processDates(payloads: Partial<Record<string, any>>[], action: Action): Partial<Record<string, any>>[] {
-		const fieldsInCollection = Object.entries(this.schema.collections[this.collection]!.fields);
+	processDates(
+		fieldEntries: [string, FieldOverview][],
+		payloads: Partial<Record<string, any>>[],
+		action: Action,
+		aliasMap: Record<string, string> = {},
+	): Partial<Record<string, any>>[] {
+		for (const alias in aliasMap) {
+			const aliasedField = aliasMap[alias];
+			const field = this.schema.collections[this.collection]!.fields[aliasedField!];
+
+			if (field) {
+				fieldEntries.push([
+					alias,
+					{
+						...field,
+						field: alias,
+					},
+				]);
+			}
+		}
 
-		const dateColumns = fieldsInCollection.filter(([_name, field]) =>
-			['dateTime', 'date', 'timestamp'].includes(field.type),
-		);
+		const dateColumns = fieldEntries.filter(([_name, field]) => ['dateTime', 'date', 'timestamp'].includes(field.type));
 
-		const timeColumns = fieldsInCollection.filter(([_name, field]) => {
+		const timeColumns = fieldEntries.filter(([_name, field]) => {
 			return field.type === 'time';
 		});
```

Comment:
As we are changing these lines do we want to optimize it a bit?
```suggestion
		const dateColumns: [string, FieldOverview][] = [];
		const timeColumns: [string, FieldOverview][] = [];

		fieldEntries.forEach((fieldEntry) => {
			if (['dateTime', 'date', 'timestamp'].includes(fieldEntry[1].type)) {
				dateColumns.push(fieldEntry);
			} else if (fieldEntry[1].type === 'time') {
				timeColumns.push(fieldEntry);
			}
		});
```

Reply:
I noticed that you often want to optimize code through restructuring and I wanted to give you an example on why this is not nearly as straight forward as it seems in JS and why you should not bother to optimize JS unless you probable cause that something is running slow and a bottle neck in the first place.

Here is code that tries to mimic the functionality from above.
```js
let data = {};

function set_data(size) {
    data = {};
    for (let i = 0; i < size; i++) {
        data[i] = { a: Math.random() > 0.5, b: Math.random() < 0.5 };
    }
    data = Object.entries(data);
}

function two_loops() {
    const as = data.filter(([_key, val]) => val.a);
    const bs = data.filter(([_key, val]) => val.b);
    return { as, bs };
}

function one_loop() {
    const as = [], bs = [];
    
    data.forEach(([_key, val]) => {
        if (val.a) { as.push([_key, val]);
        } else if (val.b) { bs.push([_key, val]); }
    });
    return { as, bs };
}

for (let size of [100, 1000, 10_000, 100_000, 1_000_000, 10_000_000]) {
    set_data(size); console.log(`Running with size ${size}`);
    console.time("one-loop");  one_loop();  console.timeEnd("one-loop");
    console.time("two-loops"); two_loops(); console.timeEnd("two-loops");
}
```
Here are the numbers of a some runs:
```
Running with size 100
one-loop:  0.186ms,  0.184ms, 0.18ms
two-loops: 0.101ms, 0.101ms, 0.099ms
Running with size 1000
one-loop:  0.195ms, 0.196ms, 0.183ms
two-loops: 1.086ms, 0.667ms, 1.062ms
Running with size 10000
one-loop:  3.511ms, 2.302ms, 2.373ms
two-loops: 2.798ms, 2.458ms, 2.734ms
Running with size 100000
one-loop: 25.675ms, 20.389ms, 24.454ms
two-loops: 4.068ms,  7.357ms,  5.628ms
Running with size 1000000
one-loop:  46.668ms, 91.161ms, 89.397ms
two-loops: 48.972ms, 51.546ms, 54.824ms
Running with size 10000000
one-loop:  579.875ms, 1001ms,    1046ms
two-loops: 677.899ms,  499.073ms, 488.883ms
```
As you can see, there is no significant difference between the two, in some runs the one-loop seems to be faster, in some the two loops. And taking into account that we're talking about fields here, it's highly unlikely that we will be working with more than 10.000 fields in a single collection.

TLDR: We should focus on maintainability and not performance unless there is probable cause for a bottleneck. JS is just not a good language for writing performant code.

Reply:
I appreciate the feedback, in the future I'll test my proposed optimization first to make sure it provides the performance I expect before suggesting it.

While I agree we should not optimize code at the expense of readability, I disagree that we should not optimize js where we can. In this case, I don't see much (if any) impact on readability with the proposed change and believe we should shave off what we can from the overall processing time. 

That being said, this is not a blocking request, so I'll go ahead and approve the PR. I'll leave it up to you if you’d like to add the optimization (modified version per your testing).